### ISSUE TITLE
Disable recursion under Linux

### ISSUE THREAD

**Issue by sebbASF** [guard/listen#556]
Does not seem possible to disable recursion under Linux.
It should be possible to watch a single directory, without any subdirectories.

**Comment by ColinDKelley** [comment]
@sebbASF I don't think this is a feature of the underlying drivers like fs-event, is it?

It's not too hard to skip over the changes you don't want in the callbacks, is it?

**Comment by ColinDKelley** [comment]
> but why not allow this to be optionally done by the library?

I was thinking to keep this gem uncomplicated. But if you want to propose an interface that allows that, please throw a proposal together.

Also, just to confirm, it sounds like you agree that this isn't a feature of the underlying drivers like `fs-event`?

**Comment by ColinDKelley** [comment]
> Now inotify does not support recursion directly, so rb-inotify emulates this if the :recursive flag is provided.
> By default it looks like listen includes the :recursive flag when configuring rb-inotify.

Ok, that sounds like a good reason to allow recursion to be disabled. If we support that, we'll need to disable it on the other drivers too. How difficult does that seem?
